### ISSUE TITLE
No way to check function declaration naming

### ISSUE THREAD

**Issue by spalger** [palantir/tslint#3998]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.8.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
function snake_case() {}
const also_snake_case = function () {}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "variable-name": {
      "options": ["check-format"]
    }
  }
}
```

#### Actual behavior

Only the variable assignment is checked.

#### Expected behavior

I personally expected function declarations to be included in the variable-name check, but since TSLint has separate checks for classes, interfaces, and variables I guess it might be desirable to add another rule for function definitions? Or do others think this would make more sense as a part of the variable-name rule?

**Comment by ajafff** [comment]
You might want to have a look at the [`naming-convention` rule of my package `tslint-consistent-codestyle`](https://github.com/ajafff/tslint-consistent-codestyle/blob/master/docs/naming-convention.md#examples). That rules allow fine grained configuration of the names of (almost) all declarations.

**Comment by JoshuaKGoldberg** [comment]
This is also in [`tslint-microsoft-contrib`](https://github.com/microsoft/tslint-microsoft-contrib)'s `function-name` rule. Merging into #1142 for bringing that rule or `naming-convention` here.
